packages/design-system/src/components/OcResourceIcon/OcResourceIcon.vue

```diff
@@ -4,24 +4,27 @@
     :name="iconName"
     :color="iconColor"
     :size="size"
-    :class="[
-      'oc-resource-icon',
-      { 'oc-resource-icon-file': !isFolder, 'oc-resource-icon-folder': isFolder }
-    ]"
+    :class="['oc-resource-icon', iconTypeClass]"
   />
 </template>
 
-<script>
+<script lang="ts">
+import { defineComponent, PropType } from 'vue'
+import { Resource } from 'web-client'
```

Comment:
Really love that you're using the `Resource` interface here. Finally that's possible! But please add `"web-client": "npm:@ownclouders/web-client"` to the list of `peerDependencies` of the design system (package.json).